### ISSUE TITLE
Dropout in OPT embedding layer

### ISSUE THREAD

**Issue by shijie-wu** [huggingface/transformers#18844]
### System Info

main

### Who can help?

@ArthurZucker, @patrickvonplaten, @LysandreJik

### Information

- [X] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

The [OPT paper](https://arxiv.org/pdf/2205.01068.pdf) (sec 2.2) mentioned 
> We use a dropout of 0.1 throughout, but we do not apply any dropout to embeddings.

This is also supported by loading [the official checkpoints](https://github.com/facebookresearch/metaseq/tree/main/projects/OPT) and running the following check

```python
import torch
data = torch.load("reshard-model_part-0.pt")
assert data['cfg']['model'].no_emb_dropout
```

However, in `modeling_opt.py`, dropout is applied to the embedding layer.

https://github.com/huggingface/transformers/blob/80367cd1fb6d36ca6bdd99b70586aab4ffae1ae1/src/transformers/models/opt/modeling_opt.py#L641-L642

### Expected behavior

No dropout in the embedding layer
